python/ray/data/dataset.py

```diff
@@ -248,17 +270,27 @@ def transform(block: Block) -> Iterable[Block]:
                 yield output_buffer.next()
 
         plan = self._plan.with_stage(
-            OneToOneStage("map", transform, compute, ray_remote_args)
+            OneToOneStage(
+                "map",
+                transform,
+                compute,
+                ray_remote_args,
+                fn=fn,
+            )
         )
         return Dataset(plan, self._epoch, self._lazy)
 
     def map_batches(
         self,
-        fn: Union[CallableClass, Callable[[BatchType], BatchType]],
+        fn: BatchUDF,
         *,
         batch_size: Optional[int] = 4096,
         compute: Union[str, ComputeStrategy] = None,
         batch_format: str = "native",
+        fn_args: Optional[List[Any]] = None,
```

Comment:
Does this need to be a Tuple per Python's convention?

Reply:
Yep good catch!

Reply:
Ah actually this can be any iterable, but I thought that it'd be easiest for the user if it was a list.